pkgs/os-specific/linux/trelay/default.nix

```
@@ -0,0 +1,45 @@
+{ stdenv, lib, fetchFromGitHub, kernel, kmod }:
+let
+    v = "22.03.5";
+in
+stdenv.mkDerivation rec {
+  name = "trelay";
+  version = "${v}-${kernel.version}";
+
+  src = fetchFromGitHub {
+    owner = "openwrt";
+    repo = "openwrt";
+    rev = "v${v}";
```

```suggestion
{ stdenv, lib, fetchFromGitHub, kernel, kmod }:
let
  version = "22.03.5";
in
stdenv.mkDerivation {
  name = "trelay";
  version = "${version}-${kernel.version}";

  src = fetchFromGitHub {
    owner = "openwrt";
    repo = "openwrt";
    rev = "v${version}";
```
we should use sparseCheckout to not download the entire repo